jmbitcoin/jmbitcoin/bip21.py

```diff
@@ -0,0 +1,41 @@
+# https://github.com/bitcoin/bips/blob/master/bip-0021.mediawiki
+# bitcoin:<address>[?amount=<amount>][?label=<label>][?message=<message>]
+# We don't check validity of Bitcoin address here, as all the tools using
+# this are expected to do address validation independently anyway.
+
+from jmbitcoin import amount_to_sat
+from urllib.parse import parse_qs, urlparse
+from url_decode import urldecode
+import re
+
+
+def is_bip21_uri(uri):
+    parsed = urlparse(uri)
+    return parsed.scheme == 'bitcoin' and parsed.path != ''
+
+
+def is_bip21_amount_str(amount):
+    return re.compile("^[0-9]{1,8}(\.[0-9]{1,8})?$").match(amount) != None
```

Comment:
Yes this seems to work, I wonder should we disallow matches to 0 though?

(And matches above 21000000? :) jk only)

Reply:
Probably add several numbers in a test case.

Reply:
That will complicate code here, sendpayment itself checks for both dust limit and having enough funds in a wallet. Added some tests for incorrectly formatted amounts.

Reply:
Yes that's fair, agreed.